sycl/include/sycl/ext/intel/experimental/esimd/simd_view.hpp

```diff
@@ -20,15 +20,15 @@ namespace intel {
 namespace experimental {
 namespace esimd {
 
-/// The reference class.
+/// The simd_view base class.
 ///
 /// This class represents a region applied to a base object, which
 /// must be a simd object.
 ///
 /// \ingroup sycl_esimd
-template <typename BaseTy, typename RegionTy> class simd_view {
+template <typename BaseTy, typename RegionTy> class simd_view_impl {
```

Comment:
this class should be moved to the detail namespace

Reply:
The `simd_view_impl` class defines all the public interfaces of `simd_view` class. Do we really want to hide it?

Reply:
yes, because users are not supposed to ever use simd_view_impl

Reply:
Right. Then I think we should also move simd_view_impl class into a separate header file, e.g. `detail/simd_view_impl.hpp`.

Reply:
sounds good

Reply:
Done.